static/src/javascripts/projects/commercial/modules/hosted/next-video-autoplay.js

```diff
@@ -0,0 +1,73 @@
+define([
+    'bean',
+    'fastdom',
+    'common/modules/analytics/omniture',
+    'common/utils/$'
+], function (
+    bean,
+    fastdom,
+    omniture,
+    $
+) {
+    var nextVideoInterval;
+    var $hostedNext = $('.js-hosted-next-autoplay');
+    var $timer = $('.js-autoplay-timer');
+    var nextVideoPage = $timer.length && $timer.data('next-page');
+
+    function cancelAutoplay() {
+        fastdom.write(function () {
+            $hostedNext.addClass('hosted-slide-out');
+        });
+        clearInterval(nextVideoInterval);
+    }
+
+    function cancelAutoplayMobile($hostedNext) {
+        fastdom.write(function () {
+            $hostedNext.addClass('u-h');
+        });
+    }
+
+    function triggerAutoplay(getCurrentTimeFn, duration) {
+        nextVideoInterval = setInterval(function () {
+            var timeLeft = duration - Math.floor(getCurrentTimeFn());
+            var countdownLength = 10; //seconds before the end when to show the timer
+
+            if (timeLeft <= countdownLength) {
+                fastdom.write(function () {
+                    $hostedNext.addClass('js-autoplay-start');
+                    $timer.text(timeLeft + 's');
+                });
+            }
+            if (timeLeft <= 0){
```

Comment:
There is `Math.floor` used a few lines above so probably not... But I just want to be extra safe ;)